writer.go

```diff
@@ -284,6 +297,65 @@ func (w *GenericWriter[T]) File() FileView {
 	return w.base.File()
 }
 
+// ConcurrentRowGroupWriter is a row group writer that can be used to write row groups
+// in parallel with other row groups. However, they must be committed back to the writer
+// serially using CommitRowGroup.
+//
+// See BeginRowGroup for more information on how this can be used.
+//
+// While multiple row groups can be created concurrently, a single row group must be created
+// sequentially.
+type ConcurrentRowGroupWriter interface {
+	RowWriterWithSchema
+	Flush() error
```

Comment:
Do you think the fall to `Flush` could be used to replace `CommitRowGroup`?

Reply:
No I do want to force pages to be flushed before committing so that final page compression can happen in parallel. I mean I guess someone could still do that with flush on each column, but I like the convenience function.

Are you talking about wanting to have the Commit method defined here to avoid the ugly cast in the current commit implementation?

Reply:
Agreed, that Commit method on the writer isn't the ideal API.

Do we need Commit or can it just be Close?

I was suggesting `Flush` because the current implementation of `Flush` is to create a row group:
```go
func (w *writer) flush() error {
	_, err := w.writeRowGroup(w.currentRowGroup, nil, nil)
	return err
}
```
There is no way to "flush pages" without materializing a row group:
1. pages are encoded and compressed when the buffers reach the configured size (might be stored on disk in temp files) -- there is no way to do this explicitly
2. flushing the temp buffers holding completed pages always creates a row group in the final output io.Writer

Reply:
Okay @achille-roussel I've pushed a change that moves the Commit method from the writers to the ConcurrentRowGroupWriter interface... PTAL